docs/docs/kds-configuration.md

```diff
@@ -54,147 +46,102 @@ enclaveHost.start(null, null, null, kdsConfiguration) {}
+When a request for a key is received by the KDS, the KDS attests that the key policy constraint matches the details of the requesting enclave before releasing a derived key.
+Just like Conclave attestation, it follows the [enclave constraints DSL](constraints.md).
+
+### Setting the `useOwnCodeHash` and `useOwnCodeSignerAndProductID` properties
+In addition to the `constraint` property, the `policyConstraint` section has two properties which allow you to include the code hash or code signer and product ID of the current enclave to the constraint.
+These parameters cannot be added to the `constraint` field directly as their values are not known at build time.
+
+- The `useOwnCodeHash` property, when set to `true`, will restrict access to the key so that it is only released to enclaves with a code hash that matches the code hash of the current enclave.
+- The `useOwnCodeSignerAndProductID` property, when set to `true`, will restrict access to the key so that it is only released to enclaves with the same code signer and product ID as the current enclave.
+
+### Example `PolicyConstraint` configurations
+#### Exact version of an enclave running on a secure platform
+This is an example of a restrictive policy constraint, whereby access to the derived key is restricted to release mode enclaves with a specific code hash running on a fully patched and up-to-date SGX capable host.
 
 ```groovy
 kds {
-  persistenceKeySpec {
-    policyConstraint {
-      constraint = "SEC:SECURE"
-      useOwnCodeHash = true
+    persistenceKeySpec {
+        policyConstraint {
+            constraint = "SEC:SECURE"
+            useOwnCodeHash = true
+        }
     }
-  }
 }
 ```
 
-### Exact version of an enclave running on a platform that may need updating
-This example is similar to above, but allows the key to be provided to an
-enclave running on a platform where a security update is available, as long as
-the platform has not been flagged as insecure. This will allow the service that
-uses the enclave to continue running whilst giving the system administrator time
-to deploy the patch to the system, bringing it back to the `SECURE` state.
+This is a restrictive policy constraint with significant operational implications.
+Because of the code hash requirement, persisted data will be inaccessible to subsequent versions of the enclave.
+Finally, if the system is not fully patched, the derived key will not be accessible and the enclave may be denied access to storage keys until patches are installed.
+
+#### Exact version of an enclave running on a platform that may need updating
+This is an example of a slightly less restrictive policy constraint, whereby access to the derived key is permitted in cases where non-critical security patches are available for the host system.
```

Comment:
In this slightly less restrictive policy example, access to the derived key is permitted in cases where non-critical security patches are available for the host system.